modules/oe_webtools_media/tests/src/FunctionalJavascript/MediaSourceWebtoolsTest.php

```diff
@@ -102,6 +103,58 @@ public function testMediaWebtoolsSource(): void {
     }
   }
 
+  /**
+   * Tests the webtools media source support for wcloud.
+   */
+  public function testMediaWebtoolsSourceWcloud(): void {
+    $session = $this->getSession();
+    $page = $session->getPage();
+    $assert_session = $this->assertSession();
+
+    // Create a webtools media type for charts.
+    $this->createWebtoolsMediaType('test_media_webtools_type', 'chart');
+
+    // Create a webtools wcloud media item without a url.
+    $this->drupalGet("media/add/test_media_webtools_type");
+    $name = "Valid webtools wcloud item";
+    $assert_session->fieldExists('Name')->setValue($name);
+    $assert_session->fieldExists("Webtools Chart snippet")->setValue('{"utility": "wcloud"}');
+    $page->pressButton('Save');
+    $assert_session->pageTextContains("The provided wcloud url is not valid.");
+
+    // Create a webtools wcloud media item with an invalid url.
+    $assert_session->fieldExists("Webtools Chart snippet")->setValue('{"utility": "wcloud", "url": "not-a-url"}');
+    $page->pressButton('Save');
+    $assert_session->pageTextContains("The provided wcloud url is not valid.");
+
+    // Create a webtools wcloud media item with a url
+    // outside the europa.eu domain.
```

Comment:
Add test case for miao.europa.eunot.com, for an empty json, for a 4xx or 5xx response.